code/game/gamemodes/cult/narsie.dm

```diff
@@ -1,362 +1,181 @@
+/obj/effect/narsie/Move()
+	. = ..()
+	if(.)
+		var/datum/extension/universally_visible/univis = get_extension(src, /datum/extension/universally_visible)
+		univis.refresh()
 
-/obj/singularity/narsie/Process()
-	eat()
+/obj/effect/narsie/forceMove(atom/dest)
+	. = ..()
+	if(.)
+		var/datum/extension/universally_visible/univis = get_extension(src, /datum/extension/universally_visible)
+		univis.refresh()
+
+/obj/effect/narsie/Process()
+
+	// Feed.
+	var/consumed = 0
+	for(var/atom/consuming in range(consume_pull, src))
+		if(get_dist(src, consuming) >= consume_range)
+			consuming.singularity_pull(src, 9)
+		else
+			consuming.on_defilement()
+		if(consumed++ >= max_atoms_assessed_per_tick || TICK_CHECK)
+			break
+
+	if(TICK_CHECK)
+		return
 
+	// Find a target.
 	if (!target || prob(5))
-		pickcultist()
-
-	move()
-
-	if (prob(25))
-		mezzer()
-
-/obj/singularity/narsie/large/eat()
-	for (var/turf/A in orange(consume_range, src))
-		consume(A)
-
-/obj/singularity/narsie/mezzer()
-	for(var/mob/living/carbon/M in oviewers(8, src))
-		if(M.stat == CONSCIOUS)
-			if(M.status_flags & GODMODE)
-				continue
-			if(!iscultist(M))
-				to_chat(M, "<span class='danger'> You feel your sanity crumble away in an instant as you gaze upon [src.name]...</span>")
-				M.apply_effect(3, STUN)
-
-
-/obj/singularity/narsie/large/Bump(atom/A)
-	if(!cause_hell) return
-	if(isturf(A))
-		narsiewall(A)
-	else if(istype(A, /obj/structure/cult))
-		qdel(A)
-
-/obj/singularity/narsie/large/Bumped(atom/A)
-	if(!cause_hell) return
-	if(isturf(A))
-		narsiewall(A)
-	else if(istype(A, /obj/structure/cult))
-		qdel(A)
-
-/obj/singularity/narsie/move(var/force_move = 0)
-	if(!move_self)
-		return 0
-
-	var/movement_dir = pick(global.alldirs - last_failed_movement)
-
-	if(force_move)
-		movement_dir = force_move
-
-	if(target && prob(60))
-		movement_dir = get_dir(src,target)
-
-	spawn(0)
-		step(src, movement_dir)
-	spawn(1)
+		var/list/targets = list()
+		var/current_zs = SSmapping.get_connected_levels(z, include_lateral = FALSE) // can't really handle lateral connections
+		var/decl/special_role/cult = GET_DECL(/decl/special_role/cultist)
+		for(var/datum/mind/cult_nh_mind in cult.current_antagonists)
+			if(cult_nh_mind.current && !cult_nh_mind.current.stat &&  (get_z(cult_nh_mind.current) in current_zs))
+				targets += cult_nh_mind.current
+		// If we have no valid cultists, go for any human.
+		if(!length(targets))
+			for(var/mob/living/carbon/human/food in global.living_mob_list_)
+				if(food.stat)
+					var/turf/pos = get_turf(food)
+					if(pos?.z in current_zs)
+						targets += food
+			// Go for ghosts if nothing else is available.
+			if(!length(targets))
+				for(var/mob/observer/G in global.dead_mob_list_)
+					targets += G
+		// Pick a random candidate.
+		if(length(targets))
+			target = pick(targets)
+		var/capname = uppertext(name)
+		to_chat(target, SPAN_NOTICE("<b>[capname] HAS LOST INTEREST IN YOU.</b>"))
```

Comment:
Is this supposed to give both 'lost interest' and 'hungers for your soul' messages to the same dude?